Classes/IndexQueue/FrontendHelper/UserGroupDetector.php

```diff
@@ -138,6 +137,12 @@ public function getPage_preProcess(
      * @param array $pageInput Page record
      * @param int $lUid Overlay language ID
      * @param PageRepository $parent Parent \TYPO3\CMS\Core\Domain\Repository\PageRepository object
+     * @see https://github.com/TYPO3-Solr/ext-solr/issues/4151
+     * @see \ApacheSolrForTypo3\Solr\EventListener\PageIndexer\SetAccessGrantedEventListener
+     *
+     * This code is not invoked in version 12 - instead an PSR-14 event was introduced to handle
+     * this case. In version 13 this has been fixed - so leaving the code here for now.
+     * The hook invoking part has been removed in this class since it makes no sense
```

Comment:
What do you mean with "in version 13 has been fixed"?
Why not removing that method and interface from that class?

Reply:
On main and version 13 it's solved via:

 #[AsEventListener]
 public function getPageOverlay_preProcess()
 
 So it's automatically listening to the event